### DISCUSSION TITLE
라우팅

### DISCUSSION THREAD

**Discussion by sujin-park0607** [24tngus/CS_STUDY#36] [Network]
# 라우팅
> 목차
> 1. 개념
>> 1.1 라우팅
>> 1.2 라우터
>> 1.3 라우팅 테이블
> 2. 게이트웨이
> 3. 홉
> 4. 실습

### 1.1 라우팅
- 네트워크에서 데이터를 보낼 때 최적의 경로를 선택하는 과정이며, 라우터가 이를 수행한다.
- 데이터는 보통 출발지에서 목적지로 가는 동안 여러개의 라우터를 거치며 여러번의 라우팅을 수행한다.
- ex) 출발지- 목적지를 가장 빠른 최적의 경로를 찾는 것

### 1.2 라우터 
- 네트워크 사이에서 데이터를 전달하는 장치이며 보통 둘 이상의 서로 다른 네트워크로 연결.
- 최적의 경로를 결정하고 경로가 결정되면 해당 경로로 데이터를 넘겨주는 일(라우팅)을 수행
- 라우터는 라우팅테이블을 기반으로 데이터를 목적지에 전달
- OSI 3계층인 네트워크의 장비이다.

### 1.3 라우팅 테이블

- IP주소를 기반으로 라우터의 위치를 저장한 테이블 & 데이터베이스
- 다양한 네트워크에 대한 정보와 해당 네트워크에 연결하는 방법이 포함되어 있음
- “네트워크에 연결하는 방법이 포함되어있는 리스트구나”
![image](https://github.com/24tngus/CS_STUDY/assets/75667075/d75d5eee-9c59-4f85-8f86-32bcc9897305)

### 1.4 예시
**상황**: PC1 → PC3로 패킷 보내기 
![image](https://github.com/24tngus/CS_STUDY/assets/75667075/ff7ef3ff-1845-4bfb-acdc-ca79df6887a0)
- **네트워크 대상**: 목적지 네트워크의 IP주소
    - 101.25.67.7이 있는 네트워크의 주소가 101.25.67.0이다
- **서브넷 마스크**: 대상 주소를 설명할 때 쓰이는 값
- **게이트 웨이**: 이 장치와 연결되어있는 홉, 패킷이 전달되는 다음 IP주소(외부 네트워크와 연결된 장치)
    - “10.0.0.4 → 외부네트워크”에서 외부네트워크에 연결된 장치가 게이트웨이이다. 그 게이트웨이의 IP가 10.0.0.2이다.
- **인터페이스**: 게이트웨이로 가기위해 거치는 장치
    - 10.0.0.2는 eth3을 통해 접근 가능
- **메트릭**: 우선순위, 패킷 전송을 위해 최적의 경로가 선택되도록 참고되는 값
    - 홉수, 지연시간,처리량 등 포함
    - 메트릭이 낮은 값이 먼저 선택돼서 가게 된다.

## 게이트웨이
- 프로토콜 변환기라고도 하며 **네트워크와 네트워크를 잇는 장치**, 라우터같은 기능

## 홉
- 네트워크에서 **출발지와 목적지 사이에 위치한 장치**를 의미하며 홉 카운트(hop count)는 데이터가 출발지와 목적지 사이에서 통과해야 하는 홉의 개수를 의미한다.
![image](https://github.com/24tngus/CS_STUDY/assets/75667075/2c1e8d4c-c197-48b4-878a-d13a6f90dc0e)
> 이것의 홉 카운트는? → 2개
> 홉카운트는 적은것이 좋다
> 메트릭에 참고된다 → 가장 최적의 경로를 구해야하기 때문에

## 실습 - 라우팅 테이블 확인하기

```java
netstat -r
```
>  라우팅 테이블 외에도 포트번호나 네트워크 상태 등 여러가지를 확인할 수 있는 명령어이기 때문에 외워두시는 것이 좋습니다! 

![image](https://github.com/24tngus/CS_STUDY/assets/75667075/aaa5767c-c6fe-4a8a-bbe5-aaaa6de22296)

> 게이트웨이
연결됨 → 로컬 게이트웨이
ip주소 → 외부게이트웨이 IP주소

**Comment by jindream6128** [comment]
확인했습니다 !

**Comment by 24tngus** [comment]
확인했습니다

**Comment by berryberries** [comment]
확인했습니다!
